content/docs/details/scaffolder/index.md

```diff
@@ -0,0 +1,385 @@
+
+# Components of a Template
+You can configure a Scaffolder _template_ to be triggered on demand by Roadie users. Your template contains one or more `steps` which run sequentially during execution of a template. You may optionally specify all `steps` while the template execution is running inside an ephemeral container that is destroyed after the execution completes.
+
+A Scaffolder template is a configurable process that will run one or more Scaffolder `steps`. The template will be run when a user visits the "Create Component" page in Backstage. `https://<tenant-name>.roadie.so/create`.
+
+![create-a-new-component](./create-a-new-component.png)
+
+Templates are defined by a Backstage Entity YAML file with a `Template` kind and imported into the Backstage catalog. You can create multiple templates, each of which can perform a different set of steps. For example, you can have one template that creates a React application, and another that creates a serverless app.
+
+Here is an example of a very basic Scaffolder template that prompts the user for a name, and then prints back the text "Hello, name!"
+
+```yaml
+apiVersion: scaffolder.backstage.io/v1beta3
+kind: Template
+metadata:
+  name: hello-world-template
+  title: Hello World
+  description: Says Hello to a specified name.
+spec:
+  owner: backstage/techdocs-core
+  type: service
+
+  parameters:
+    - title: You are about to say hello to your first Backstage Template
+      required:
+        - name
+      properties:
+        name:
+          title: Name
+          type: string
+          description: Name to say hello to
+          ui:autofocus: true
+          ui:options:
+            rows: 5
+
+  steps:
+    - id: log-message
+      name: Log Message
+      action: debug:log
+      input:
+        message: 'Hello, ${{ parameters.name }}!'
+```
+
+# Header Section
+
+The header section is required for every `template` and contains information to configure the task and show details about the task on the "Create Component" page.
+
+```yaml
+apiVersion: scaffolder.backstage.io/v1beta3
+kind: Template
+metadata:
+  name: hello-world-template
+  title: Hello World
+  description: Says Hello to a specified name.
+spec:
+  owner: default/engineering
+  type: service
+```
+
+## `apiVersion`
+This is a required field and should be set to `scaffolder.backstage.io/v1beta3`
+
+## `kind`
+A Scaffolder template is also an Entity in Backstage. In order to configure this entity as a template you must set the kind to `Template`
+
+
+## `metadata`
+The metadata field contains some data that appears on the template card that appears on the "Create Component" page.
+
+## `spec`
+The spec field contains `owner` and `type`. Owner refers to the Backstage group or user that owns the Scaffolder task e.g. `default/engineering`. Type refers to the type of template. It can be set to anything and appears on the scaffolder template card in the "Create Component" page.
+
+# `parameters`
```

Comment:
I find it confusing that "Header Section" (above) is a H1 title, whereas "parameters" is a code snippet in a H1. I would avoid using inline code in the header.

Reply:
yeah hmmm. I did this specifically for the side bar. Maybe we can do a design on this.

Reply:
Look at this page: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions

Reply:
Fair enough!